lib/modules/platform/gerrit/index.ts

```diff
@@ -0,0 +1,543 @@
+}
+
+async function findOwnPr(
+  findPRConfig: GerritFindPRConfig,
+  refreshCache?: boolean
+): Promise<GerritChange[]> {
+  const filterState = mapPrStateToGerritFilter(findPRConfig.state);
+  const filter = ['owner:self', 'project:' + config.repository!, filterState];
+  if (findPRConfig.branchName !== '') {
+    filter.push(`hashtag:sourceBranch-${findPRConfig.branchName}`);
+  }
+  if (findPRConfig.targetBranch) {
+    filter.push(`branch:${findPRConfig.targetBranch}`);
+  }
+  if (findPRConfig.label) {
+    filter.push(`label:Code-Review=${findPRConfig.label}`);
+  }
+  const changes = await client.findChanges(filter, refreshCache);
+  logger.debug(`findOwnPr(${filter.join(', ')}) => ${changes.length}`);
+  return changes;
+}
+
+export async function findPr(
+  findPRConfig: FindPRConfig,
+  refreshCache?: boolean
+): Promise<Pr | null> {
+  const change = (await findOwnPr(findPRConfig, refreshCache)).pop();
+  return change ? mapGerritChangeToPr(change) : null;
+}
+
+export async function getPr(number: number): Promise<Pr | null> {
+  try {
+    const change = await client.getChange(number);
+    return Promise.resolve(mapGerritChangeToPr(change));
+  } catch (err) {
+    if (err.statusCode === 404) {
+      return Promise.resolve(null);
+    }
+    throw err;
+  }
+}
+
+export async function updatePr(prConfig: UpdatePrConfig): Promise<void> {
+  logger.debug(`updatePr(${prConfig.number}, ${prConfig.prTitle})`);
+  const change = await client.getChange(prConfig.number);
+  if (change.subject !== prConfig.prTitle) {
+    await updatePullRequestTitle(
+      prConfig.number,
+      change.change_id,
+      prConfig.prTitle
+    );
+  }
+  if (prConfig.prBody) {
+    await updatePullRequestBody(prConfig.number, prConfig.prBody);
+  }
+  if (prConfig.platformOptions?.autoApprove) {
+    await client.approveChange(prConfig.number);
+  }
+  if (prConfig.state && prConfig.state === 'closed') {
+    await client.abandonChange(prConfig.number);
+  }
+}
+
+export async function createPr(prConfig: CreatePRConfig): Promise<Pr | null> {
+  logger.debug(
+    `createPr(${prConfig.sourceBranch}, ${prConfig.prTitle}, ${
+      prConfig.labels?.toString() ?? ''
+    })`
+  );
+  const pr = (
+    await findOwnPr(
+      {
+        branchName: prConfig.sourceBranch,
+        targetBranch: prConfig.targetBranch,
+        state: 'open',
+      },
+      true
+    )
+  ).pop();
+  if (pr) {
+    //Workaround for "Known Problems.1"
+    if (pr.subject !== prConfig.prTitle) {
+      await updatePullRequestTitle(pr._number, pr.change_id, prConfig.prTitle);
+    }
+    await updatePullRequestBody(pr._number, prConfig.prBody);
+    if (prConfig.platformOptions?.autoApprove) {
+      await client.approveChange(pr._number);
+    }
+    return getPr(pr._number);
+  } else {
+    throw new Error(
+      `the change should be created automatically from previous push to refs/for/${prConfig.sourceBranch}`
+    );
+  }
+}
+
+async function updatePullRequestTitle(
```

Comment:
can you move all those (not exported) helper functions to a seperate file because this file is already huge?